frontend/server/express.mjs

```diff
@@ -535,26 +600,30 @@ app.get("/is-dev", async(req, res) => {
 
   app.post("/new-block-react", (req, res) => {
     const data = req.body;
-
+  
+    // Function to unescape special characters in the script
     function unescape(s) {
       return s
         .replace(/\\n/g, "\n")
         .replace(/\\r/g, "\r")
         .replace(/\\"/g, '"')
         .replace(/\\\\/g, "\\");
     }
-
+  
     let folderPath = data.block_name;
     const filePath = path.join(data.blockPath, "computations.py");
-    const computations_script = unescape(data.computations_script);
-
+  
+    // Prepend a comment with the chat history index to the computations script
+    const computations_script = `# Chat history index: ${data.chatHistoryIndex}\n` + unescape(data.computations_script);
```

Comment:
It feels flaky to me to store app state in a user file. wouldn't it make more sense to store it in the chatHistory file?